generator/nextjs/template/src/components/iota/IotaClientPage.tsx

```diff
@@ -29,111 +30,94 @@ type DataRequests = {
-  const [configOptions, setConfigOptions] = useState<SelectOption[]>([]);
-  const [selectedConfig, setSelectedConfig] = useState<string>("");
-  const [iotaSession, setIotaSession] = useState<Session>();
-  const [iotaIsInitializing, setIotaIsInitializing] = useState(false);
-  const [queryOptions, setQueryOptions] = useState<SelectOption[]>([]);
+  const [selectedConfigId, setSelectedConfigId] = useState<string>("");
   const [selectedQuery, setSelectedQuery] = useState<string>("");
   const [openMode, setOpenMode] = useState<OpenMode>(OpenMode.Popup);
   const [dataRequests, setDataRequests] = useState<DataRequests>({});
   const [isFormDisabled, setIsFormDisabled] = useState(false);
 
-  // Prefill did from session
+  // Get did from session
   const { data: session } = useSession();
-  useEffect(() => {
-    if (!session || !session.user) return;
-    setHolderDid(session.userId);
-  }, [session]);
 
-  useEffect(() => {
-    const initConfigurations = async () => {
-      try {
-        const response = await fetch("/api/iota/configuration-options", {
-          method: "GET",
-        });
-        const configurations = await response.json();
-        setConfigOptions(configurations);
-      } catch (error) {
-        console.error("Error getting Iota configurations:", error);
-      }
-    };
-    if (featureAvailable) {
-      initConfigurations();
-    }
-  }, [featureAvailable]);
+  const configurations = useQuery({
+    queryKey: ["configurations"],
+    queryFn: fetchIotaConfigurations,
+    enabled: !!featureAvailable,
+  });
 
-  async function handleConfigurationChange(value: string | number) {
-    const configId = value as string;
-    clearSession();
-    setSelectedConfig(configId);
-    if (!configId) {
-      return;
-    }
-    try {
-      setIotaIsInitializing(true);
-      getQueryOptions(configId);
-      const credentials = await getIotaCredentials(configId);
+  const iotaSession = useQuery({
+    queryKey: ["iotaSession", selectedConfigId],
+    queryFn: async ({ queryKey }) => {
+      const credentials = await getIotaCredentials(queryKey[1]);
       const iotaSession = new Session({ credentials });
       await iotaSession.initialize();
-      setIotaSession(iotaSession);
-    } catch (error) {
-      if (error instanceof IotaError) {
-        console.log(error.code);
-      }
-    } finally {
-      setIotaIsInitializing(false);
-    }
-  }
+      return iotaSession;
+    },
+    enabled: !!selectedConfigId,
+  });
 
-  async function getQueryOptions(configurationId: string) {
-    const response = await fetch(
-      "/api/iota/query-options?" +
-      new URLSearchParams({
-        iotaConfigurationId: configurationId,
-      }),
-      {
-        method: "GET",
-      },
-    );
-    const options = (await response.json()) as SelectOption[];
-    setQueryOptions(options);
-  }
+  const queryOptions = useQuery({
+    queryKey: ["configurations", selectedConfigId],
```

Comment:
Just making sure I understand correctly. This uses the a composite key of `configurations` and the `selectedConfigId` to store the queries? Why not call the first key something like `queries` or `configurationQueries` instead of just `configurations`?

Reply:
Good idea! Changed naming of queries in both pages 👍

Reply:
Oh, I meant the name of the key, not the name of the constant
`queryKey: ["configurations"],`

Reply:
My bad! I completely misread what you wrote. I do like the new variable names better so no harm 😅 

Regarding query keys, if not a dependency, they are just used for managing query caching: https://tanstack.com/query/v4/docs/framework/react/guides/query-keys. 

After searching around, it seems like general convention is to name them after what data is being fetched in the query, thus I think the current naming makes sense. The one thing I did find is that they should be unique across the whole app, so I renamed `configurations` -> `iotaConfigurations`/`issuanceConfigurations`.

Let me you if you disagree with any of this!

Reply:
This one is not storing configurations, its storing queries

Reply:
Was also updated `configurations` -> `queryOptions` as part of the change above!

(I'm used to Gitlab updating referenced code in comment chains, but apparently GIthub doesn't do that so I forgot to mention it, sorry! 🙇 )